### PR TITLE
[Logs] Be more lenient when filtering containers using labels

### PR DESCRIPTION
**Pull request by mnshdw**
### What does this PR do?

This PR changes the way we filter Docker container when there is a `label` in the configuration.

Previously, we took the whole label and checked whether there was a container that had that label as a key.

With this change, we now accept a comma separated list of labels that can be formatted either as `key:value`, `key=value`, or simply `key`.

### Motivation

This should address https://github.com/DataDog/datadog-agent/issues/946.

### CODE REVIEW

File: pkg/logs/input/container/scanner.go [hunk 1]
```diff
@@ -122,12 +123,24 @@ func (s *Scanner) listContainers() []types.Container {
...
+		// Expect a comma-separated list of labels, eg: foo:bar, baz
+		for _, value := range strings.Split(source.Label, ",") {
+			// Trim whitespace, then check whether the label format is either key:value or key=value
+			label := strings.TrimSpace(value)
+			parts := strings.FieldsFunc(label, func(c rune) bool {
+				return c == ':' || c == '='
+			})
+			// If we have exactly two parts, check there is a container label that matches both.
+			// Otherwise fall back to checking the whole label exists as a key.
+			if _, ok := container.Labels[label]; ok || len(parts) == 2 && container.Labels[parts[0]] == parts[1] {
```
Comment (ajacquemot) [inline pkg/logs/input/container/scanner.go]: I'd rather use `exists` than `ok`

**Review by ajacquemot — COMMENTED** [review]
This looks good to me, I just left a tiny comment. I am wondering if Kenafeh has not implemented this feature already, did you check on your side ?

File: pkg/logs/input/container/scanner.go [hunk 2]
```diff
@@ -122,12 +123,24 @@ func (s *Scanner) listContainers() []types.Container {
 }
 
 func (s *Scanner) sourceShouldMonitorContainer(source *config.IntegrationConfigLogSource, container types.Container) bool {
-	if source.Image != "" && container.Image != source.Image {
-		return false
+	if source.Image != "" {
```
Comment (tmichelet) [inline pkg/logs/input/container/scanner.go]: Why are we changing this?

**Review by tmichelet — CHANGES_REQUESTED** [review]
One regression, otherwise 👍

**Review by ajacquemot — APPROVED** [review]

**Review by tmichelet — APPROVED** [review]

**Pull request merged**